cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/transaction/coprocessor/hbase11/DefaultTransactionProcessor.java

```diff
@@ -83,55 +87,91 @@ public void stop(CoprocessorEnvironment e) throws IOException {
+    if (maxLifetimeFromConf == null) {
+      // handle the case where txMaxLifetime was never initialized from cConf
+      if (txMaxLifetimeMillis == null) {
+        // Try to get the value from Operation
         byte[] maxLifetimeBytes = op.getAttribute(HBaseTable.TX_MAX_LIFETIME_MILLIS_KEY);
         if (maxLifetimeBytes != null) {
-          maxLifetimeMillis = Bytes.toLong(maxLifetimeBytes);
+          return Bytes.toLong(maxLifetimeBytes);
         } else {
-          LOG.warn("txMaxLifetimeMillis is not available in client's operation attributes. " +
-                     "Defaulting to default tx_max_lifetime");
-          maxLifetimeMillis = TimeUnit.SECONDS.toMillis(TxConstants.Manager.DEFAULT_TX_MAX_LIFETIME);
+          // If maxLifetimeMillis could not be found in the client request as well, just use the default value
+          long defaultMaxLifeTimeInSecs = TxConstants.Manager.DEFAULT_TX_MAX_LIFETIME;
+          LOG.debug(String.format("txMaxLifetimeMillis is not available in client's operation attributes. " +
+                                    "Defaulting to the default value of %d seconds for region %s.",
+                                  defaultMaxLifeTimeInSecs, env.getRegionInfo().getRegionNameAsString()));
+          return TimeUnit.SECONDS.toMillis(defaultMaxLifeTimeInSecs);
         }
+      } else {
+        // If we have a value from the previous check, return that
+        return txMaxLifetimeMillis;
       }
     } else {
-      maxLifetimeMillis = txMaxLifetimeMillis;
+      // If conf has the value, update the variable and return it
+      txMaxLifetimeMillis = maxLifetimeFromConf;
+      return maxLifetimeFromConf;
     }
+  }
 
-    boolean validLifetime =
-      TxUtils.getTimestamp(tx.getTransactionId()) + maxLifetimeMillis > System.currentTimeMillis();
-    if (!validLifetime) {
-      throw new DoNotRetryIOException(String.format("Transaction %s has exceeded max lifetime %s ms",
-                                                    tx.getTransactionId(), maxLifetimeMillis));
+  private void reloadPruneState(RegionCoprocessorEnvironment env) {
+    if (pruneEnable == null) {
+      // If prune enable has never been initialized, try to do so now
+      initializePruneState(env);
+    } else {
+      Configuration conf = getConfiguration(env);
+      if (conf != null) {
+        boolean newPruneEnable = conf.getBoolean(TxConstants.TransactionPruning.PRUNE_ENABLE,
+                                                 TxConstants.TransactionPruning.DEFAULT_PRUNE_ENABLE);
+        if (newPruneEnable != pruneEnable) {
+          // pruning enable has been changed, resetting prune state
+          if (LOG.isDebugEnabled()) {
+            LOG.debug(String.format("Transaction Invalid List pruning feature is set to %s now for region %s.",
+                                    newPruneEnable, env.getRegionInfo().getRegionNameAsString()));
+          }
+          resetPruneState();
```

Comment:
There is no guarantee that resetting prune state will propagate the changes to prune upper bound writer. Since the reset happens only on a compaction or a flush.